.github/workflows/publish-release.yml

```diff
@@ -3,7 +3,14 @@ name: Publish Release
 on:
   pull_request:
```

Comment:
So in order for this to publish correctly, it needs to be triggered by the `main` branch. We can adjust the triggers here to something more like:

```
on:
  push:
    branches: [main]
...

Then later we can check to see if each commit is a release like this:
```
  is-release:
    # Filtering by `push` events ensures that we only release from the `main` branch, which is a
    # requirement for our npm publishing environment.
    # The commit author should always be 'github-actions' for releases created by the
    # 'create-release-pr' workflow, so we filter by that as well to prevent accidentally
    # triggering a release.
    if: github.event_name == 'push' && startsWith(github.event.head_commit.author.name, 'github-actions')
    needs: all-jobs-pass
    outputs:
      IS_RELEASE: ${{ steps.is-release.outputs.IS_RELEASE }}
    runs-on: ubuntu-latest
    steps:
      - uses: MetaMask/action-is-release@v1
        id: is-release
```

and in `publish-release`, filter by `IS_RELEASE`

```
  publish-release:
    needs: is-release
    if: needs.is-release.outputs.IS_RELEASE == 'true'
```